tools/test.py

```diff
@@ -44,13 +49,49 @@ def run_tests(tests, bin_path, test_args=[], algorithms=[]):
                 print(" ".join(test_cmd))
                 subprocess.run(test_cmd, check=True, shell=True, env=my_env)
 
+                if coverage:
+                    generate_report(test_name, my_env)
+
+
+def generate_report(test, env):
+    print(f"Generating coverage report for {test}.")
+    subprocess.run(["mkdir", "-p", f"coverage/{test}/html"], env=env)
+    subprocess.run([
+        "llvm-profdata",
+        "merge",
+        "-sparse",
+        f"{test}.profraw",
+        "-o",
+        f"coverage/{test}/{test}.profdata"
+    ], env=env)
+    with open(f"coverage/{test}/{test}.lcov", 'wb') as lcov_file:
+        subprocess.run([
+            "llvm-cov",
+            "export",
+            "-format",
+            "lcov",
+            "--instr-profile",
+            f"coverage/{test}/{test}.profdata",
+            test,
+            "../../src",
+        ], stdout=lcov_file, env=env)
+    subprocess.run([
+        "genhtml",
```

Comment:
It needs to be documented that we use lcov and that it needs to be installed for coverage analysis.

Reply:
I described it in the README 👍🏻